src/prefect/_result_records.py

```diff
@@ -232,4 +232,4 @@ def deserialize_from_result_and_metadata(
     def __eq__(self, other: Any | "ResultRecord[Any]") -> bool:
         if not isinstance(other, ResultRecord):
             return False
-        return self.metadata == other.metadata and self.result == other.result
+        return self.metadata == other.metadata and self.result == other.result  # type: ignore[reportUnknownVariableType]
```

Comment:
Maybe we can flip the conditional to appease `pyright`?
```python
if isinstance(other, ResultRecord):
    return self.metadata == other.metadata and self.result == other.result
else:
    return False
```

Reply:
hmm that didn't work, but a `model_dump(include` did!